services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -1,11 +1,148 @@
+      this.evaluateTransactionsAgainstDynamicRules.bind(this);
+  }
+
+  async runInlineRule(projectId: string, inlineRule: InlineRule) {
+    const evaluateFn = this._evaluateNameToFunction[inlineRule.fnName];
+    if (!evaluateFn) {
+      throw new Error(`No evaluation function found for rule name: ${inlineRule.id}`);
+    }
+
+    return await evaluateFn({
+      ...inlineRule.options,
+      projectId,
+    });
+  }
+
+  async evaluateTransactionsAgainstDynamicRules({
+    projectId,
+    amountThreshold,
+    amountBetween,
+    direction = 'inbound',
+    excludedCounterpartyIds = [],
+    paymentMethods = [],
+    excludePaymentMethods = false,
+    timeAmount = 7,
+    timeUnit = 'days',
+    groupByBusiness = false,
+    groupByCounterparty = false,
+    havingAggregate = AggregateType.SUM,
+  }: TransactionsAgainstDynamicRulesType) {
+    const conditions: Prisma.Sql[] = [
+      Prisma.sql`"projectId" = ${projectId}`,
+      Prisma.sql`"transactionDirection"::text = ${direction}`,
+      Prisma.sql`"transactionDate" >= CURRENT_DATE - INTERVAL '${this.getIntervalTime(
+        timeUnit,
+        timeAmount,
+      )}'`,
+    ];
+
+    if (excludedCounterpartyIds.length) {
+      const conditions = excludedCounterpartyIds
+        .map(id => `counterpartyOriginatorId NOT LIKE '${id}'`)
+        .join(' AND ');
+
+      conditions.concat(conditions);
+    }
+    if (paymentMethods.length) {
+      const methodCondition = excludePaymentMethods ? `NOT IN` : `IN`;
+      conditions.push(
+        Prisma.sql`"paymentMethod"::text ${Prisma.raw(methodCondition)} (${Prisma.join(
+          paymentMethods,
+        )})`,
+      );
+    }
+
+    if (amountBetween) {
+      conditions.push(
+        Prisma.sql`"transactionAmount" BETWEEN ${amountBetween.min} AND ${amountBetween.max}`,
+      );
+    }
+
+    // build conditional select, businessId alone, counterpartyOriginatorId alone, or both
+    let selectClause: Prisma.Sql;
+    let groupByClause: Prisma.Sql;
+    if (groupByBusiness) {
+      selectClause = Prisma.sql`"businessId"`;
+      conditions.push(Prisma.sql`"businessId" IS NOT NULL`);
+      groupByClause = Prisma.raw(`"businessId"`);
+    } else if (groupByCounterparty) {
+      selectClause = Prisma.sql`"counterpartyOriginatorId"`;
+      conditions.push(Prisma.sql`"counterpartyOriginatorId" IS NOT NULL`);
+      groupByClause = Prisma.raw(`"counterpartyOriginatorId"`);
+    } else {
+      selectClause = Prisma.sql`"businessId", "counterpartyOriginatorId"`;
+      groupByClause = Prisma.raw(`"businessId", "counterpartyOriginatorId"`);
+      conditions.push(Prisma.sql`"businessId" IS NOT NULL`);
+      conditions.push(Prisma.sql`"counterpartyOriginatorId" IS NOT NULL`);
+    }
+
+    const whereClause = Prisma.join(conditions, ' AND ');
+
+    let havingClause: string = '';
+    switch (havingAggregate) {
+      case AggregateType.COUNT:
+        havingClause = `${AggregateType.COUNT}(id)`;
+        break;
```

Comment:
Same spacing for separating return statements from code before. 🙏🏼 
```suggestion
        havingClause = `${AggregateType.COUNT}(id)`;
        
        break;
```